### PR TITLE
onnx: add the Flatten operator.

### PR DESCRIPTION
**Pull request by wanglong001**
onnx: add the Flatten operator.

ref: https://github.com/onnx/onnx/blob/8a6b70f0adcff5409b8cd65883bbbea7d6c4f4e1/onnx/reference/ops/op_flatten.py#L11

### CODE REVIEW

File: candle-onnx/src/eval.rs [hunk 1]
```diff
@@ -766,6 +766,20 @@ pub fn simple_eval(
...
                 values.insert(node.output[0].clone(), output);
             }
+            //  https://github.com/onnx/onnx/blob/main/docs/Operators.md#flatten
+            "Flatten" => {
+                let axis = get_attr_opt::<i64>(node, "axis")?.copied().unwrap_or(1) as usize;
+                let input = get(&node.input[0])?;
+                let end_index = input.shape().dims().iter().product::<usize>();
+                let new_shape = if axis == 0 {
+                    (1, end_index)
+                } else {
```
Comment (LaurentMazare) [inline candle-onnx/src/eval.rs:776]: What about the case where axis would not be 1?

Comment (wanglong001) [reply candle-onnx/src/eval.rs:776]: Hi， Referencing this link https://github.com/onnx/onnx/blob/main/docs/Operators.md#flatten, in the case where axis > 1, it is compatible with axis == 1.

Comment (LaurentMazare) [reply candle-onnx/src/eval.rs:776]: Couldn't the case with axis set to 0 be merged with the other case? The product from `first_part` would be `1` on an empty iterator?

Comment (wanglong001) [reply candle-onnx/src/eval.rs]: you are right. I made changes.

**Review by LaurentMazare — APPROVED** [review]

**Comment by LaurentMazare** [comment]
Thanks!

**Pull request merged**